### PR TITLE
Tooltip | FE | Refactor and redesign

### PR DESCRIPTION
**Pull request by mikemai2awesome**
## Jira

http://vjira2:8080/browse/BDS-1461

## Summary

Introduces a new design and accessibility improvements for the Tooltip component.

## Details

1. Twig and JS are completely rewritten from scratch, so hell yeah, breaking changes.
2. Twig has logic to accept legacy data structure.
3. Description is rewritten to emphasize the difference between tooltip (floating label) and popover (small overlay).
4. `trigger` now accepts any kind of content and proper `aria-` attributes are in place.
5. Screen readers can now correctly announce the trigger and the tooltip content.
6. `placement` is added to introduce all 12 different placements the tooltip content can have. In addition, auto placement will place the tooltip content based on available space on the screen.

## Deprecated

Again, THERE ARE BREAKING CHANGES. The twig file is configured to accept legacy data structure but the following props are deprecated:

* `transform`
* `direction`
* `noWrap`
* `trigger.toggle`

All existing tooltips should be inspected and converted to use the proper component:
* **tooltip** is a floating label that briefly describes an element.
* **popover** is a small overlay that contains structured information and calls to action.

## How to test

Pull down the branch locally and perform the following.

### Check tests

- [ ] 1. Given I have pulled down the branch locally
- [ ] 2. And I am in the root folder
- [ ] 3. When I run the command `npx jest packages/components/bolt-tooltip/__tests__/tooltip.js`
- [ ] 4. Then I should see all tests indicated as "Pass"

### Check hover interaction

- [ ] 1. Given I am on the tooltip [doc page](https://feature-bolt-tooltip-refactor.boltdesignsystem.com/pattern-lab/patterns/02-components-tooltip/index.html)
- [ ] 2. And I am using the mouse
- [ ] 3. When I hover the mouse over a tooltip trigger
- [ ] 4. Then I should see tooltip content to appear in a floating label

### Check keyboard interaction

- [ ] 1. Given I am on the tooltip [doc page](https://feature-bolt-tooltip-refactor.boltdesignsystem.com/pattern-lab/patterns/02-components-tooltip/index.html)
- [ ] 2. And I am using the keyboard to navigate
- [ ] 3. When I tap to a tooltip trigger
- [ ] 4. Then I should see tooltip content to appear in a floating label

### Check screen reader interaction

- [ ] 1. Given I am on the tooltip [doc page](https://feature-bolt-tooltip-refactor.boltdesignsystem.com/pattern-lab/patterns/02-components-tooltip/index.html)
- [ ] 2. And I am using the keyboard to navigate
- [ ] 3. When I tap to a tooltip trigger
- [ ] 4. Then I should hear the trigger being announced as an expanded button
- [ ] 5. And I should hear the tooltip content's text getting announced

### CODE REVIEW

**Comment by mikemai2awesome** [comment]
@danielamorse That would happen in every browser if the tooltip isn't placed correctly, it's not IE specific.

File: packages/components/bolt-link/link.schema.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+// @todo: WIP
```
Comment (adamszalapski) [inline packages/components/bolt-link/link.schema.js]: Why are we changing bolt-link in bolt-tooltip PR?

Comment (mikemai2awesome) [reply packages/components/bolt-link/link.schema.js]: I forgot about this one. I believe we needed the JS schema to make it work, but we didn't change anything else so link is unaffected by this PR. We can finish converting link to use JS schema later. Correct, @danielamorse?

Comment (danielamorse) [reply packages/components/bolt-link/link.schema.js]: Yes, not necessary for this PR. Save the work somewhere!

**Comment by sghoweri** [comment]
PR was released with v2.15.0

**Comment by sghoweri** [comment]
PR was released with v2.15.0

**Pull request merged**